plugins/woocommerce/src/Blocks/Templates/AbstractTemplate.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * AbstractTemplate class.
+ *
+ * Shared logic for templates.
+ *
+ * @internal
+ */
+abstract class AbstractTemplate {
+
+	/**
+	 * The slug of the template.
+	 *
+	 * @var string
+	 */
+	const SLUG = '';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->init();
+	}
+
+	/**
+	 * Initialization method.
+	 */
+	abstract public function init();
```

Comment:
Why do we need to make this one public? AFAIK, it's only called from the constructor.